### ISSUE TITLE
Support OAuth 2.0 device flow

### ISSUE THREAD

**Issue by DrDaveD** [hashicorp/vault-plugin-auth-jwt#103]
I suggest that this plugin optionally support the [OAuth 2.0 device flow](https://tools.ietf.org/html/draft-ietf-oauth-device-flow-15) for authentication.  This avoids needing to have a callback to a web server on the client's machine and is much easier to manage from the command line.

**Comment by DrDaveD** [comment]
This would probably be based on this [unmerged golang oauth2 pull request](https://github.com/golang/oauth2/pull/356) or this [add-on package](https://github.com/rjw57/oauth2device).

**Comment by kalafut** [comment]
Hi. Thanks for the references. Support for Device Flow is definitely of interest, especially for remote login support. It hasn’t been investigated in detail and there isn’t an ETA at this time, but it’s something that will be looked into.

**Comment by DrDaveD** [comment]
I am beginning to work on a pull request for this.

**Comment by kalafut** [comment]
Hi @DrDaveD. Can you summarize the approach you plan to take?

**Comment by sidewinder12s** [comment]
Was able to work around this by adding this flag when SSHing into a remote machine:

```
ssh -L 8250:localhost:8250 <address>
```

And then clicking on the Auth URL when it was presented.

**Comment by tsipinakis** [comment]
4 years later pinging this issue again as it's a much desired feature for us - it breaks a lot of workflows to not be able to log in easily on SSH (if you forward the port then you can't log in locally).

**Comment by DrDaveD** [comment]
The port of #131 is now available in an [openbao 2.1.0 beta release](https://github.com/openbao/openbao/releases/tag/v2.1.0-beta20241114.3) via  openbao/openbao#319.
